### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by peva3** [blockai/torrent-http-bridge#1]
Really like the idea! But just to clarify one thing for me, would a usecase for this be essentially taking a URL file, lets say a .iso file and "bridging" it to be distributed from webtorrent?

If so, that could be a really interesting way to run a sort of distributed CDN or something like that.

Thanks!

**Comment by olalonde** [comment]
Yes exactly :). It takes any file hosted on an http server (URL) and makes the file available for download through bittorrent. We're using this as part of https://github.com/blockai/bitstore-client.

There are a few limitations at the moment: it assumes the torrent file already exists (it won't download the remote file locally and generate the torrent file at the moment) and I'm still trying to figure out how to enable webrtc support on the server and it's a PITA to debug.

**Comment by peva3** [comment]
Is there no way to create the .torrent file on the fly when someone "selects" the file they want to distribute? Then the magnetic link gets passed somehow to webtorrent. And have some sort of index so if the file is already in the system the user just downloads the already created torrent?

**Comment by olalonde** [comment]
That would be possible but I haven't implemented this right now for 2 reasons:

1) I didn't need to because I'm only using this to seed URLs for which I've already generated the torrent file

2) doing so is possible but requires temporarily downloading the entire file locally to generate its piece hashes... I'm open to pull requests though (as long as we keep this optional) :)

Eventually, I'd like to reduce the memory footprint of running bitstore-torrent for a large amount of files by keeping its state in a database and only loading torrent objects into memory when at least one peer starts downloading. I'm not sure how feasible that is though, I suppose it would likely involve modifying https://github.com/feross/bittorrent-swarm/blob/master/lib/tcp-pool.js.
